File: Práctica 1/Java/IrrgartenP1/src/irrgarten/Weapon.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author aloncraftmc
 */
public class Weapon {
    
    private float power;
    private int uses;
    
    public Weapon(float power, int uses){
        
        this.power = power;
        this.uses = uses;
        
    }
    
    public float attack(){
        
        if(this.uses <= 0)  return 0;
        
        this.uses --;
        return this.power;
        
    }
    
    public boolean discard(){
        return Dice.discardElement(this.uses);
    }
    
    public String toString(){
        return "W[" + this.power + ", " + this.uses + "]";
    }
    
}

File: Práctica 1/Ruby/weapon.rb
# encoding: UTF-8

module Irrgarten

    class Weapon

        def initialize(power, uses)

            @power = power
            @uses = uses

        end

        def attack

            return 0 if @uses <= 0

            @uses -= 1
            return @power

        end

        def discard
            return Dice.discard_element(@uses)
        end

        def to_s
            return "W[" + @power.to_s + ", " + @uses.to_s + "]"
        end

    end

end